build-support/bin/black.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import argparse
+import subprocess
+from typing import List
+
+
+def create_parser() -> argparse.ArgumentParser:
+  parser = argparse.ArgumentParser(
+    description="Formats all python files since master with black through pants.",
+  )
+  parser.add_argument("-f", "--fix", action="store_true",
+    help="Instead of erroring on files with incorrect formatting, fix those files."
+  )
+  return parser
+
+
+def main() -> None:
+  args= create_parser().parse_args()
+  if args.fix:
+    black_args: List[str] = []
+  else:
+    black_args = ["--black-check"]
```

Comment:
Can replace these 4 lines with one line for a ternary expression:

```python
black_args = ["--black-check"] if args.fix else []
```

Reply:
(I went a slightly different route because without shell = True, the extra empty array caused an issue)